findArrayDepth.js

```diff
@@ -0,0 +1,16 @@
+const findArrayDepth = (array) => {
+  let maxLevel = 0;
+  function findDepth(arr, level) {
+    if (Array.isArray(arr)) {
+      maxLevel = Math.max(maxLevel, level);
+      arr.forEach((element) => findDepth(element, level + 1));
+    }
+  }
+  findDepth(array, 0);
+  return maxLevel;
+}
+
+const flattenArray = (array) => {
+    const arrayDepth = findArrayDepth(array);
+    return array.flat(arrayDepth);
+}
```

Comment:
Again it is not good to create a variable that you will only use once e.g: `arrayDepth`